make csharpcore work.

bagnier opened the pull request:
- [ X] I acknowledge that this PR is not a solution to the Gilded Rose Kata, but an improvement to the template.

## Please provide your PR description below this line

This PR makes csharpcore work. 

I add a missing NuGet, use only one test framework, downgrade to stick to NET6 LTS and var some variable declarations.

Tested on Visual Studio Mac and Visual Studio 2022 for Windows.

this resolves #465

emilybache commented:
ok so the trouble is I'm on .net 7.0 so this doesn't work for me... and I also prefer to provide both NUnit and XUnit as options if possible. Do you think you could copy these changes to a new folder called csharpcore_6LTS or similar? Then people who are on .net 7.0 can carry on as before?

bagnier commented:
Hey !

my bad, I should do more atomic changes. I reverted the .NET version to restore 7. Tested on Mac but not yet on Windows.

I could create several folders for each .NET version but I'm not sure of the value vs complexity. 

I removed xunit because it seemed to me misleading to mix 2 test frameworks, one for the foo test and another for the smoke test. It would make sens to create two folders, one based on xUnit and another on NUnit.

Thank you for sharing and maintaining this awsome kata ! thrilled to read and follow your next recommandations.

emilybache commented:
Hi, I think this addresses my concerns about the .net version and I think most people can convert themselves to xunit if they prefer it. Thanks!
